acoustid/future/fpindex/updater.py

```
@@ -0,0 +1,172 @@
+    """Load initial fingerprint data using a consistent snapshot"""
+
+    async with AsyncExitStack() as exit_stack:
+        await exit_stack.enter_async_context(
+            conn.transaction(isolation="repeatable_read", readonly=True)
+        )
+
+        needs_import, _ = await create_replication_slot(conn)
+        if not needs_import:
+            return
+
+        async def drop_replication_slot_on_error(*exc_details) -> None:
```

Ruff (0.8.2)

Missing type annotation for `*exc_details`

(ANN002)